library/src/main/java/com/joom/grip/GripFactory.kt

```diff
@@ -16,17 +16,25 @@
 
 package com.joom.grip
 
+import com.joom.grip.commons.mapToArray
 import com.joom.grip.io.IoFactory
 import com.joom.grip.mirrors.ReflectorImpl
-import org.objectweb.asm.Opcodes
 import java.io.File
+import java.nio.file.Path
 import javax.annotation.concurrent.ThreadSafe
+import org.objectweb.asm.Opcodes
 
 @ThreadSafe
 interface GripFactory {
+  @Deprecated("Deprecated", ReplaceWith("createFromPath(file, files)"))
   fun create(file: File, vararg files: File): Grip
+
+  @Deprecated("Deprecated", ReplaceWith("createFromPath(files)"))
   fun create(files: Iterable<File>): Grip
 
+  fun createFromPath(path: Path, vararg paths: Path): Grip
+  fun createFromPath(paths: Iterable<Path>): Grip
```

Comment:
It would be nice to provide default implementations and compile the library with `-Xjava-default=all`.